### ISSUE TITLE
Random NetworkRetrievalError for OCSP when running Get-EnterprisePKIHealthStatus

### ISSUE THREAD

**Issue by PChipVIA** [PKISolutions/PSPKI#216]
For some reason, randomly the Get-EnterprisePKIHealthStatus returns a NetworkRetrievalError on the OCSP.  However the Enterprise PKI and even testing with OpenSSL do not return any errors.   

Environment Information:
Powershell version 5.1.14393.6343.  Also the same with 5.1.19041.4412.
PSPKI version 4.2.0 (latest with fix #215)

```
PS C:\> Get-EnterprisePKIHealthStatus -CertificateAuthority (Get-CertificationAuthority server.******.tld) | Select -ExpandProperty Childs | Select -f 1 -ExpandProperty URLs


Name              : AIA Location #1
Status            : Ok
ExtendedErrorInfo :
Url               : http://pki1.******.ca/PKI-12(1).crt
ExpirationDate    : 2/2/2029 8:46:28 AM
UrlType           : Certificate

Name              : AIA Location #2
Status            : Ok
ExtendedErrorInfo :
Url               : http://pki2.******.ca/PKI-12(1).crt
ExpirationDate    : 2/2/2029 8:46:28 AM
UrlType           : Certificate

Name              : CDP Location #1
Status            : Ok
ExtendedErrorInfo :
Url               : http://pki1.******.ca/PKI-12(1).crl
ExpirationDate    : 6/3/2024 4:55:48 PM
UrlType           : Crl

Name              : CDP Location #2
Status            : Ok
ExtendedErrorInfo :
Url               : http://pki2.******.ca/PKI-12(1).crl
ExpirationDate    : 6/3/2024 4:55:48 PM
UrlType           : Crl

Name              : OCSP Location #1
Status            : NetworkRetrievalError
ExtendedErrorInfo :
Url               : http://ocsp.******.ca/ocsp
ExpirationDate    :
UrlType           : Ocsp

```

**Comment by PChipVIA** [comment]
Got one

```
PS C:\> $xchg = Get-CAExchangeCertificate -X509 -CertificationAuthority (Get-CA $([System.Net.DNS]::GetHostEntry('').HostName))
PS C:\> $ocspReq = New-Object SysadminsLV.PKI.OcspClient.OCSPRequest $xchg
PS C:\> $ocspReq.SendRequest()

Exception calling "SendRequest" with "0" argument(s): "Input data does not represent valid 'GeneralizedTime' type."
At line:1 char:1
+ $ocspReq.SendRequest()
+ ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : Asn1InvalidTagException
 

PS C:\> 
```

**Comment by Crypt32** [comment]
Ok, so the problem is in ASN.1 parser which has some issues with parsing GeneralizedTime. Can you do one more test, please:
```PowerShell
$rsp = Invoke-WebRequest -Uri "http://ocsp.domain.ca/ocsp/MFQwUjBQME4wTDAJBgUrDgMCGgUABBQ2zpAvKWOUMw2afYilvaVl8dQIyQQUM4u/nCATOKj9kEr40DNXzdbTIZgCEyAAAMxsBtSL03bA/GgAAQAAzGw="
# ensure that yo get HTTP 200 response
$rsp
# this will produce a binary response
[convert]::ToBase64String($rsp.Content)
```
And post this base64 string. Alternatively (for privacy concerns), you can email it to me: vpodans&sysadmins.lv (replace ampersnad with @). This will provide me enough information to find exact issue and fix it.

**Comment by Crypt32** [comment]
The fix will be available in v4.3
